compose_rl/utils/utils.py

```diff
@@ -1084,3 +1085,12 @@ def make_action_mask(
         mask[i, :prompt_len[i] - 1] = 0
 
     return mask
+
+
+def flatten(coll: Union[Iterable[Any], str]) -> Generator[Any, None, None]:
```

Comment:
doc str
(cuz i'm pedantic)
use this one or do you

```suggestion
def flatten(coll: Union[Iterable[Any], str]) -> Generator[Any, None, None]:
    """Recursively flattens an arbitrarily nested iterable (excluding strings).

    Note: strings are treated as atomic elements and are not flattened into
    characters.

    Args:
        coll (Union[Iterable[Any], str]): The nested iterable to flatten.

    Yields:
        Any: The individual, non-iterable elements from the flattened structure.
    """
```